3723/1/top_gems.rb

```diff
@@ -0,0 +1,38 @@
+require_relative 'repo'
+require_relative 'pages'
+require 'optparse'
+
+options = {}
+
+OptionParser.new do |parser|
+  parser.on('-t', '--top=NUMBER') do |number|
+    options[:number] = number
+  end
+  parser.on('-n', '--name=NAME') do |name|
+    options[:name] = name
+  end
+  parser.on('-f', '--file=FILE') do |file|
+    options[:file] = file
+  end
+end.parse!
+
+doc = if options[:file]
```

Comment:
I don't say that it's better, but I'd like to discuss which is better and why, your current statement or something like:
```ruby
doc_path = option[:file] || 'gems.yml'
doc = YAML.load_file(doc_path)
```
or even
```ruby
doc = YAML.load_file(options[:file] || 'gems.yml')
```

Reply:
Done